subsys/dfu/src/dfu_target_modem.c

```diff
@@ -171,6 +171,9 @@ int dfu_target_modem_init(size_t file_size, dfu_target_callback_t cb)
 	if (offset == DIRTY_IMAGE) {
 		delete_banked_modem_fw();
 	} else if (offset != 0) {
+#if defined(CONFIG_DFU_TARGET_ALWAYS_RESTART)
```

Comment:
Could you have done this in `dfu_target.c` instead? So that instead of invoking the `..modem` and `...mcuboo` variants you handle it at a higher level.

Reply:
`dfu_target.c` can't set the internal offset state of the `dfu_target_modem` of-course I can expose it through an additional function `dfutarget_set_offset` do you think that's better?